source/game.gamescriptexpression.bmx

```diff
@@ -675,23 +676,28 @@ Type TGameScriptExpression extends TGameScriptExpressionBase
 				tv = TNewsEventTemplate(context.context).templateVariables
 			EndIf
 		EndIf
-		
-		If tv
-			result = _ParseWithTemplateVariables(variable, context, tv)
+
+		Local varLowerCase:String = variable.ToLower()
+		If tv And tv.HasVariable(varLowerCase, True)
+			result = _ParseWithTemplateVariables(varLowerCase, context, tv)
 		Else
-			result = TGameScriptExpressionBase.GameScriptVariableHandlerCB(variable, context)
+			'TODO parse Expression if it contains further variables? 
+			'${.worldtime:"year"} was resolved without further changes...
+			Local code:String = TLocalization.GetLanguageCode(localeID)
```

Comment:
Ein Vorschlag fuer hier.

Wir haben ja nun schon ein Event was auf den Sprachwechsel reagiert. Man koennte auf weitere hoeren um "Caches" zu leeren (die "nosave" markiert sind).
Caches die "Lookups" sein koennten (fuer den LanguageCode).
Aber glaube "so viel" bringt das an der Stelle nix.

Reply:
Hast Du schon ueberlegt, ob es was bringt, die Variable im Fall von "oh, ist vom GetDatabaseLocalizer() bereitgestellt" danach einfach in den templateVars hinzuzufuegen? (if tv then tv.AddVariable) ?

Dann muesste doch automatisch die Variable als "resolved" gespeichert werden (falls gewuenscht) und somit der Endwert "konstant" werden.

Habe es jetzt nicht nachgeprueft aber ich denke das momentane Vorgehen muesste immer wieder "neu" auswerten (muesste ja testbar sein, wenn die globale variable eine random-expression enthaelt)

Reply:
Ich werde auf jeden Fall eine get-Methode mit languageID einführen. Dann kann man sich später noch Gedanken um eine Optimierung machen. Statt eines Caches für die aktuelle Sprache könnte stattdessen das Sprachid-Code-Array lokal vorgehalten werden. Um IDs außerhalb des Definitionsbereichs muss man sich trotzdem kümmern. Ich würde hier erstmal nicht überoptimieren.

Auch das Hinzufügen zu den TemplateVariablen würde ich erstmal lassen. 
* Globale Variablen sollten eigentlich keine Zufallswerte enthalten.
* Man hätte bereits die Möglichkeit des Zwischenspeicherns (`<varInTemplate><all>${globalVar}</all></varInTemplate>`) analog randomcity.
* Alternativen werden nur in TemplateVariable unterstützt, nicht in allgemeinen Expressions (obwohl ein `${.persongenerator:"${de|it|us}":"firstname}` eigentlich ganz neckisch wäre, daher kann man das aktuell auch nicht direkt testen

Ich wäre dafür, die globalen Variablen überhaupt erstmal ans Rennen zu bekommen.